store/store.go

```diff
@@ -345,6 +345,101 @@ func (s *Store[H]) HasAt(_ context.Context, height uint64) bool {
 	return height != uint64(0) && s.Height() >= height
 }
 
+// DeleteTo implements [header.Store] interface.
+func (s *Store[H]) DeleteTo(ctx context.Context, to uint64) error {
+	var from uint64
+
+	if tailPtr := s.tailHeader.Load(); tailPtr != nil {
```

Comment:
Can we return an explicit error if tailHeader is not set on store for whatever reason? (unititialised store for example)

Reply:
TBH I'm not sure do we need this. After #274 store will be initialized from `.Start` so this case will be impossible(-ish, unless Store isn't started).

@Wondertan wdyt?

Reply:
We can implement this suggestion as part of Init pr